packages/primitives/popover/src/utils/cdk-event.service.ts

```diff
@@ -0,0 +1,166 @@
+        return this.primitiveConfigs?.get(primitiveInstance)?.[eventTypeAsPrimitiveConfigKey(eventType)];
+    }
+
+    addClickDomRootEventCallback(callback: (event: MouseEvent) => void) {
+        this.#clickDomRootEventCallbacks.add(callback);
+    }
+
+    removeClickDomRootEventCallback(callback: (event: MouseEvent) => void) {
+        return this.#clickDomRootEventCallbacks.delete(callback);
+    }
+
+    #setPreventPrimitiveFromCdkEvent<
+        T extends object,
+        R extends EventType,
+        K extends PrimitiveConfig[EventTypeAsPrimitiveConfigKey<R>]
+    >(primitiveInstance: T, eventType: R, value: K) {
+        if (!this.primitiveConfigs?.has(primitiveInstance)) {
+            isDevMode() &&
+                console.error(
+                    '[RdxCdkEventService.preventPrimitiveFromCdkEvent] RDX Primitive instance has not been registered!',
+                    primitiveInstance
+                );
+            return;
+        }
+        switch (eventType) {
+            case 'cdkOverlayOutsideClick':
+                this.primitiveConfigs.get(primitiveInstance)!.preventCdkOverlayOutsideClick = value;
+                break;
+            case 'cdkOverlayEscapeKeyDown':
+                this.primitiveConfigs.get(primitiveInstance)!.preventCdkOverlayEscapeKeyDown = value;
+                break;
+        }
+    }
+
+    #registerOnDestroyCallbacks() {
+        this.destroyRef.onDestroy(() => {
+            this.onDestroyCallbacks.forEach((onDestroyCallback) => onDestroyCallback());
+            this.onDestroyCallbacks.clear();
+        });
+    }
+
+    #listenToClickDomRootEvent() {
+        const target = this.document;
+        const eventName = 'click';
+        const options: boolean | AddEventListenerOptions | undefined = { capture: true };
+        const callback = (event: MouseEvent) => {
+            this.#clickDomRootEventCallbacks.forEach((clickDomRootEventCallback) => clickDomRootEventCallback(event));
+        };
+        target.addEventListener(eventName, callback, options);
+        const destroyClickDomRootEventListener = () => {
+            target.removeEventListener(eventName, callback, options);
+            this.#clickDomRootEventCallbacks.clear();
+        };
+        this.onDestroyCallbacks.add(destroyClickDomRootEventListener);
+    }
+}
+
+const RdxCdkEventServiceToken = new InjectionToken<RdxCdkEventService>('RdxCdkEventServiceToken');
+
+const existsErrorMessage = 'RdxCdkEventService should be provided only once!';
+
+const deleteRdxCdkEventServiceWindowKey = () => {
+    delete (window as any)[RdxCdkEventServiceWindowKey];
+};
+
+const getProvider: (throwWhenExists?: boolean) => Provider = (throwWhenExists = true) => ({
+    provide: RdxCdkEventServiceToken,
+    useFactory: () => {
+        isDevMode() && console.log('providing RdxCdkEventService...');
+        if ((window as any)[RdxCdkEventServiceWindowKey]) {
```

Comment:
@pimenovoleg regarding that https://github.com/orgs/radix-ng/discussions/213#discussioncomment-11637733 - can we fix all the SSR isses within the dedicated issue -> https://github.com/radix-ng/primitives/issues/202 ?

Would be nice to merge it so that I can continue on `popover`.

I rcommend make the service global, for all of the primitives that use Angular CDK (I think this should be a standard pick as we are creating a lib for Angular - it is easier and faster to use ready to go CDK funabilities instead of inventing things like overlay from scratch; in the future we could think about our own CDK 😄)

Reply:
@pimenovoleg if we decide to have it global, I will make it global and add a documentation for it.

Reply:
@pimenovoleg if we decide to have it global, I will make it global and add a documentation for it.